### PR TITLE
Remove unused functions

### PR DESCRIPTION
**Pull request by practicalswift**
Remove unused functions.

**Note to reviewers:** Perhaps some of these were meant to be called. If so, please let me know where/how :-)

### CODE REVIEW

File: gossipd/handshake.c [hunk 1]
```diff
@@ -108,20 +83,6 @@ struct act_three {
...
-	 * `1 byte` for the handshake version, `33 bytes` for the ephemeral
-	 * public key encrypted with the `ChaCha20` stream cipher, `16 bytes`
-	 * for the encrypted public key's tag generated via the `AEAD`
-	 * construction, and `16 bytes` for a final authenticating tag.
-	 */
-	BUILD_ASSERT(sizeof(act3->v) == 1);
-	BUILD_ASSERT(sizeof(act3->ciphertext) == 33 + 16);
-	BUILD_ASSERT(sizeof(act3->tag) == 16);
-}
-
```
Comment (rustyrussell) [inline gossipd/handshake.c]: These functions check that our structures match the requirements; the compilation will fail if this isn't true.

Thus, they don't need to be called, but they *do* need to exist.  Perhaps add a comment instead?

Comment (practicalswift) [reply gossipd/handshake.c]: @rustyrussell I see. I'll fix that. What would be an appropriate place to call them?

Comment (cdecker) [reply gossipd/handshake.c]: I added the calls in #429, seems we had the same idea :-)

**Comment by practicalswift** [comment]
@rustyrussell @cdecker Closing this in favor of #429. Thanks for reviewing! :-)

**Pull request closed without merging**